draft-nygren-httpbis-httpssvc.md

```diff
@@ -0,0 +1,791 @@
+
+This data type can be represented as an Unknown RR as described in
+{{!RFC3597}}:
+
+ www.example.com. 3600 IN TYPE??? \\# TBD:WRITEME
+
+This construction is intended to be extensible in two ways.  First,
+any extensions that are made to the Alt-Svc format for transmission
+over HTTPS are also applicable here, unless expressly mentioned
+otherwise.
+
+Second, by defining a way to map non-HTTPS schemes and non-default
+ports we provide a way for the HTTPSSVC to be used for them as needed.
+However, by using the origin name for the RRName for scheme https and
+port 443 we allow HTTPSSVC records to be included at the end of CNAME
+chains for existing site implementations without requiring changes in
+the zone containing the origin.
+
+
+# Differences from Alt-Svc as transmitted over HTTP
+
+Publishing an alternative services form HTTPSSVC record in DNS is
+intended to be equivalent to transmitting this field value over HTTPS,
+and receiving an HTTPSSVC record is intended to be equivalent to
+receiving this field value over HTTPS.  However, there are some small
+differences in the intended client and server behavior.
+
+## Omitting Max Age and Persist
+
+When publishing an HTTPSSVC record in DNS, server operators MUST omit the
+"ma" parameter, which encodes the "max age" (i.e. expiration time) of
+an Alt-Svc Field Value.  Instead, server operators SHOULD encode the
+expiration time in the DNS TTL, and MUST NOT set a TTL longer than the
+intended "max age".
+
+When receiving an HTTPSSVC record, clients MAY synthesize a new "ma"
+parameter from the DNS TTL, in order to interoperate with Alt-Svc processing
+subsystems.
+
+When publishing an HTTPSSVC record, server operators MUST omit the
+"persist" parameter, which indicates whether the client should use
+this record on other network paths.  When receiving an HTTPSSVC record,
+clients MUST discard any records that contain a "persist" flag.
+Disabling persistence is important to prevent an adversary on one
+network from implanting a forged DNS record that allows them to
+track users after they leave that network.
+
+## Multiple records and preference ordering
+
+Server operators MAY publish multiple HTTPSSVC records as an RRSET.  When
+publishing an RRSET with multiple HTTPSSVC
+records, the server operator MUST set the overall TTL to the minimum
+of the "max age" values (following Section 5.2 of {{!RFC2181}}).
+
+Each RR MUST contain exactly one alt-value, as described
+in Section 3 of {{!AltSvc}}.
+
+As RRs within an RRSET are explictly unordered collections, a new
+"pri" alt-svc parameter is introduced to indicate priority.  The value
+of the "pri" parameter is a non-negative integer between 0 and 65535.
+Alt-Svc alt-values with a smaller pri value should be given preference
+over alt-values with a larger pri value.
+
+TBD: what should default pri values be for Alt-Svc records
+received via DNS and HTTP when pri is unspecified?
```

Comment:
Information coming over an HTTPS connection is authenticated by the certificate, while DNS is potentially unauthenticated.  Records from an authenticated connection are always higher priority than those from DNS.

Reply:
Agreed.

Reply:
Proposal in #14.